### PR TITLE
Usergrid 323 windows phone notifications support

### PR DESCRIPTION
**Pull request by shawnfeldman**

### CODE REVIEW

File: stack/services/src/main/java/org/apache/usergrid/services/notifications/ProviderAdapterFactory.java [hunk 1]
```diff
@@ -37,6 +38,7 @@ public static ProviderAdapter getProviderAdapter(Notifier notifier, EntityManage
        switch(notifier.getProvider().toLowerCase()){
            case "apple" : adapter = new APNsAdapter(entityManager,notifier); break;
            case "google" : adapter = new GCMAdapter(entityManager ,notifier); break;
+           case "windows" : adapter = new WNSAdapter(entityManager ,notifier); break;
            case "noop" : adapter = new TestAdapter(notifier); break;
```
Comment (tnine) [inline stack/services/src/main/java/org/apache/usergrid/services/notifications/ProviderAdapterFactory.java]: Seems silly, but I think our default case should be to throw an exception if it doesn't match any of the other cases.  That way we get some explicit feedback something was entered incorrectly into the payload.

File: stack/services/src/main/java/org/apache/usergrid/services/notifications/wns/WNSAdapter.java [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+                        break;
+                    case "badge":
+                        WnsBadge badge;
+                        if (translatedNotification.getMessage() instanceof Integer) {
+                            badge = new WnsBadgeBuilder().value((Integer) translatedNotification.getMessage()).build();
+                        } else {
+                            badge = new WnsBadgeBuilder().value(translatedNotification.getMessage().toString()).build();
+                        }
+                        service.pushBadge(providerId, badge);
+                        break;
```
Comment (tnine) [inline stack/services/src/main/java/org/apache/usergrid/services/notifications/wns/WNSAdapter.java]: Same thing here, do we want to throw/log an exception for when it's not a type we handle?  That would be helpful operationally if users are having problems and can't send.

File: stack/services/src/main/java/org/apache/usergrid/services/notifications/ProviderAdapterFactory.java (hunk 1 above)
Comment (shawnfeldman) [inline stack/services/src/main/java/org/apache/usergrid/services/notifications/ProviderAdapterFactory.java]: fixed

File: stack/services/src/main/java/org/apache/usergrid/services/notifications/wns/WNSAdapter.java (hunk 2 above)
Comment (shawnfeldman) [inline stack/services/src/main/java/org/apache/usergrid/services/notifications/wns/WNSAdapter.java]: fixed

**Comment by tnine** [comment]
Looks good.

**Pull request merged**